src/psyclone/psyir/frontend/fparser2.py

```diff
@@ -105,97 +151,114 @@ def _check_bound_is_full_extent(array, dim, index, operator):
     bound Fortran code is captured as longhand lbound and/or
     ubound functions as expected in the PSyIR.
 
-    The supplied "array" argument is assumed to be an Array node,
-    the specified dimension "dim" is assumed to be a Range node
-    and the supplied range index "index" is assumed to be valid.
+    The supplied "array" argument is assumed to be an Array node and
+    the contents of the specified dimension "dim" is assumed to be a
+    Range node.
+
+    This routine is only in fparser2.py until #717 is complete as it
+    is used to check that array syntax in a where statement is for the
+    full extent of the dimension. Once #717 is complete this routine
+    can be moved into fparser2_test.py as it is used there in a
+    different context.
 
     :param array: the node to check.
     :type array: :py:class:`pysclone.psyir.node.array`
-    :param int dim: the dimension index to check.
-    :param int index: the index of the range to check (0 is the \
-        lower bound, 1 is the upper bound).
-    :param operator: the expected operator.
+    :param int dim: the dimension index to use.
+    :param operator: the operator to check.
     :type operator: \
         :py:class:`psyclone.psyir.nodes.binaryoperation.Operator.LBOUND` \
         or :py:class:`psyclone.psyir.nodes.binaryoperation.Operator.UBOUND`
 
+    :raises TypeError: if the supplied arguments are of the wrong type.
     :raises NotImplementedError: if the supplied array does not have \
```

Comment:
Not any more. Please update and document the :return: and :rtype:.

Reply:
Sorry, I missed that.